daemon/algod/api/algod.oas2.json

```diff
@@ -3972,6 +4045,74 @@
           }
         }
       }
+    },
+    "SimulationUnnamedGroupResources": {
+      "description": "If unnamed resource access is allowed, this is the subset of unnamed resources that were accessed by this group and could benefit from group resource sharing. In other words, there are no restrictions on where these resources must be placed in the transaction group. Also see property 'unnamed-resources-accessed' in SimulateTransactionResult to see unnamed resources accessed by each transaction which cannot benefit from group sharing; in contrast, these resources must be placed local to that transaction.",
+      "type": "object",
+      "required": [
+        "resources"
+      ],
+      "properties": {
+        "resources": {
+          "$ref": "#/definitions/SimulationUnnamedResourceAssignment"
+        },
+        "asset-holdings": {
+          "description": "The unnamed asset holdings that were referenced. The order of this array is arbitrary.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AssetHoldingReference"
+          }
+        },
+        "app-locals": {
+          "description": "The unnamed app local states that were referenced. The order of this array is arbitrary.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApplicationLocalReference"
+          }
+        }
+      }
+    },
+    "SimulationUnnamedResourceAssignment": {
+      "description": "The set of unnamed resources that were accessed during a simulation call.",
+      "type": "object",
+      "required": [],
+      "properties": {
+        "accounts": {
+          "description": "The unnamed accounts that were referenced. The order of this array is arbitrary.",
+          "type": "array",
+          "items": {
+            "type": "string",
+            "x-algorand-format": "Address"
+          }
+        },
+        "assets": {
+          "description": "The unnamed assets that were referenced. The order of this array is arbitrary.",
+          "type": "array",
+          "items": {
+            "type": "integer",
+            "x-algorand-format": "uint64"
+          }
+        },
+        "apps": {
+          "description": "The unnamed applications that were referenced. The order of this array is arbitrary.",
+          "type": "array",
+          "items": {
+            "type": "integer",
+            "x-algorand-format": "uint64"
+          }
+        },
+        "boxes": {
+          "description": "The unnamed boxes that were referenced. The order of this array is arbitrary.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BoxReference"
+          }
+        },
+        "empty-box-refs": {
```

Comment:
Perhaps we should report this as two integers "read-bytes-overbudget" and "write-bytes-overbudget"? It seems like "empty-box-ref" is too specific. You _could_ fix this by duplicating a box ref (non-empty) for example.  I don't know - I guess that's a dumb thing to do and we wouldn't be encouraging it.

And I guess we need to description to be very clear - this is what is needed _in addition_ to adding the box refs implied by `boxes`, right?

Reply:
Definitely open to renaming this so its purpose is more clear. Maybe `extra-box-refs` would be better.

And yes this is the number of refs in addition to the ones you'd have if you named every referenced unnamed box.

Reporting `read-bytes-overbudget` and `write-bytes-overbudget` would be sufficient replacements (assuming clients knows/can figure out the bytes per boxref), but it would require some deeper changes to get these numbers. Let me know if you feel strongly that this is a better metric to report.

Reply:
I like `extra-box-refs`. It's sufficiently non-specific about them being empty.

Reply:
Done in 9b112ee